client/app/components/queries/visualization-embed.js

```diff
@@ -22,6 +22,9 @@ const VisualizationEmbed = {
   },
 };
 
+const queryStringAsObject = () => location.search.slice(1).split('&').map(p => p.split('='))
+  .reduce((obj, [key, value]) => ({ ...obj, [key.replace(/^p_/, '')]: value }), {});
```

Comment:
😱 
We already have code in that takes parameters from the query string, we should reuse it.

Reply:
Could you help me locate the piece of code you are mentioning? I did find similar behavior when constructing a [Parameters](https://github.com/getredash/redash/blob/master/client/app/services/query.js#L214) instance, but that is also bound to the actual query text, and doesn't seem to have the built-in ability to convert to a simple key-value JSON.

Reply:
Yeah, I guess we need to consolidate this. As a quick fix there's [this](https://github.com/getredash/redash/pull/3681/files), but we should really have this logic in one place. Could you let me know which piece of code you were mentioning in this comment?